lib/network_test.go

```diff
@@ -1294,3 +1297,198 @@ func TestDecodeBlockVersion0(t *testing.T) {
 
 	require.Equal(expectedBytes, blockBytes)
 }
+
+// This test will test determinism and correctness of TransactionSpendingLimit.ToMetamaskString().
+func TestSpendingLimitMetamaskString(t *testing.T) {
+	require := require.New(t)
+	_ = require
+
+	// Number of operations to choose from during tests. The following fields should reflect the upper bound on
+	// the corresponding TransactionSpendingLimit fields.
+	maxTxnType := 26
+	maxCreatorCoinLimitOperation := 4
+	maxDAOCoinLimitOperation := 6
+	maxNFTLimitOperation := 7
+
+	// Number of random operations to generate for each field.
+	testOperationCount := 2
+
+	// We test different configurations of TransactionSpendingLimit fields.
+	// Generate a random GlobalDESOLimit field.
+	_populateTotalDESOLimit := func() uint64 {
+		return rand.Uint64()
+	}
+	// Generate a random TransactionCountLimitMap field.
+	_populateTransactionCountLimitMap := func(operationCount int) map[TxnType]uint64 {
+		operationMap := make(map[TxnType]uint64)
+
+		for ; operationCount > 0; operationCount-- {
+			txnTyp := TxnType(uint8(rand.Int()%maxTxnType + 1))
```

Comment:
you're not guaranteed to have 26 txn types with this since you may overwrite existing ones. Is this okay?

Reply:
haha okay, good find. This was me being lazy, but gotchu, it's fixed now.